rust/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -127,34 +128,38 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
         let (ism_addresses, threshold) = ism.modules_and_threshold(message).await.context(CTX)?;
         let threshold = threshold as usize;
 
-        let metas = join_all(
-            ism_addresses
-                .iter()
-                .map(|ism_address| self.base.build(*ism_address, message)),
-        )
-        .await;
-
-        let sub_modules = join_all(
+        let sub_modules_and_metas = join_all(
             ism_addresses
                 .iter()
-                .map(|ism_address| self.base.build_ism(*ism_address)),
+                .map(|ism_address| self.base.build_ism_and_metadata(*ism_address, message)),
         )
         .await;
 
-        let filtered_sub_module_metas = metas
+        // Partitions things into
+        // 1. ok_sub_modules: ISMs with metadata with valid metadata
+        // 2. err_sub_modules: ISMs with invalid metadata
+        let no_err_sub_modules_and_metas: Vec<_> = sub_modules_and_metas
             .into_iter()
-            .enumerate()
-            .zip(sub_modules.into_iter())
-            .filter_map(|((index, meta_result), sub_module_result)| {
-                match (meta_result, sub_module_result) {
-                    (Ok(Some(meta)), Ok(ism)) => Some(IsmAndMetadata::new(ism, index, meta)),
-                    _ => None,
-                }
-            })
+            .filter_map(|sub_module_and_meta| sub_module_and_meta.ok())
             .collect();
-
+        let (ok_sub_modules, err_sub_modules): (Vec<_>, Vec<_>) = no_err_sub_modules_and_metas
+            .into_iter()
+            .enumerate()
+            .partition_map(
+                |(index, sub_module_and_meta)| match sub_module_and_meta.metadata {
+                    Some(_) => Either::Left(IsmAndMetadata::new(
```

Comment:
you can do `Some(metadata) => ` and then use metadata instead of `sub_module_and_meta.metadata.unwrap()`